moveit_ros/visualization/robot_state_rviz_plugin/src/robot_state_display.cpp

```diff
@@ -404,6 +399,10 @@ void RobotStateDisplay::onDisable()
 void RobotStateDisplay::update(float wall_dt, float ros_dt)
 {
   Display::update(wall_dt, ros_dt);
+
+  if (load_robot_model_)
+    loadRobotModel();
+
```

Comment:
`RobotStateDisplay` and `TrajectoryDisplay` didn't yet have the background job functionality and I didn't want to introduce it with this PR. However, using a background job was my original idea as well (https://github.com/ros-planning/moveit/pull/234#pullrequestreview-891625). 
On the other hand, the loading in `update()` happens only once and is thus not very different from the old behavior, where loading was done once during initialization.